x-pack/plugins/observability_solution/synthetics/server/routes/monitor_cruds/add_monitor.ts

```diff
@@ -86,7 +87,16 @@ export const addSyntheticsMonitorRoute: SyntheticsRestApiRouteFactory = () => ({
         });
       }
 
-      const normalizedMonitor = validationResult.decodedMonitor;
+      const normalizedMonitor = Object.assign(
+        {},
+        validationResult.decodedMonitor,
+        await mapInlineToProjectFields({
+          monitorType: validationResult.decodedMonitor?.type,
+          monitor: validationResult?.decodedMonitor,
+          logger: server.logger,
+          includeInlineScript: false,
```

Comment:
```suggestion
          includeInlineScript: true,
```

otherwise, it won't save inline script at all in saved object

Reply:
No we'll still have the inline script because it's included in `validationResult.decodedMonitor`. This would _overwrite_ the inline script data, which is what I was trying to avoid by passing this flag. It'd replace it with an exact copy but I try not to overwrite if there's no mutation. I think the best thing to do here is change the API and get rid of the default value of `true` for this flag, and then in cases like this it can be omitted, and make the inclusion opt-in rather than opt-out.

![20241016120146](https://github.com/user-attachments/assets/d27cea04-77b6-40c6-80ff-316958cfea75)


This did lead me to discover a case where we _are_ dropping the inline script when persisting an **edit** however, so I'm going to push a fix for that now 😄